sentry-uitest/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/util/BenchmarkOperation.kt

```diff
@@ -0,0 +1,129 @@
+            op2Name: String,
+            warmupIterations: Int = 3,
+            measuredIterations: Int = 15
+        ): BenchmarkResult {
+            // The first operations are the slowest, as the device is still doing things like filling the cache.
+            repeat(warmupIterations) {
+                op1.warmup()
+                op2.warmup()
+            }
+            // Now we can measure the operations (in alternating sequence).
+            repeat(measuredIterations) {
+                op1.iterate()
+                op2.iterate()
+            }
+            val op1Result = op1.getResult(op1Name)
+            val op2Result = op2.getResult(op2Name)
+
+            // Let's print the raw results.
+            println("=====================================")
+            println(op1Name)
+            println(op1Result)
+            println("=====================================")
+            println(op2Name)
+            println(op2Result)
+            println("=====================================")
+
+            return op2Result.compare(op1Result)
+        }
+    }
+
+    private var iterations: Int = 0
+    private var durationNanos: Long = 0
+    private var cpuDurationMillis: Long = 0
+    private var frames: Int = 0
+    private var droppedFrames: Double = 0.0
+    private var lastFrameTimeNanos: Long = 0
+
+    /** Run the operation without measuring it. */
+    private fun warmup() {
+        op()
+        isolate()
+    }
+
+    /** Run the operation and measure it, updating sentry-uitest-android-benchmark data. */
+    private fun iterate() {
+        val startRealtimeNs = SystemClock.elapsedRealtimeNanos()
+        val startCpuTimeMs = Process.getElapsedCpuTime()
+
+        lastFrameTimeNanos = startRealtimeNs
+        iterations++
+        choreographer.postFrameCallback(frameCallback)
+
+        op()
+
+        choreographer.removeFrameCallback(frameCallback)
+        cpuDurationMillis += Process.getElapsedCpuTime() - startCpuTimeMs
+        durationNanos += SystemClock.elapsedRealtimeNanos() - startRealtimeNs
+
+        isolate()
+    }
+
+    /** Return the [BenchmarkOperationResult] for the operation. */
+    private fun getResult(operationName: String): BenchmarkOperationResult = BenchmarkOperationResult(
+        cpuDurationMillis / iterations,
+        droppedFrames / iterations,
+        durationNanos / iterations,
+        // fps = counted frames per seconds converted into frames per nanoseconds, divided by duration in nanoseconds
+        // We don't convert the duration into seconds to avoid issues with rounding and possible division by 0
+        (frames * TimeUnit.SECONDS.toNanos(1) / durationNanos).toInt(),
+        operationName
+    )
+
+    /**
+     * Helps ensure that operations don't impact one another.
+     * Doesn't appear to currently have an impact on the benchmark.
+     */
+    private fun isolate() {
+        Thread.sleep(500)
+        Runtime.getRuntime().gc()
```

Comment:
Curious why we need to call `gc`.

Reply:
We try to reduce the impact a benchmark run can have on the following one. We cannot clear the caches, but at least we can ask the system to destroy the objects of the previous run, and avoid they get reused.
Also, multiple runs of different operations could fill the ram, or in any case increase its usage, distorting the tests.
We try to run each test in the same conditions, as much as we can